### ISSUE TITLE
Create own EDIFACT JAXB entity

### ISSUE THREAD

**Issue by sungung** [smooks/smooks-edi-cartridge#125]
Trying to figure out how to generate JAXB entity for EDIFACT message. The reason I want to build own JAXB entities is to accomodate user defined identifier or qualifier.

Also I noticed org.smooks.edifact.binding.d95b.E1153ReferenceQualifier from out of box doesn't have all specified qualifers in the specification.

Can you please guide me how to build onw JAXB enitities from modified EDIFACT message specification?

```
@XmlType(name = "E1153-ReferenceQualifier")
@XmlEnum
public enum E1153ReferenceQualifier {

    AAA;

    public String value() {
        return name();
    }

    public static E1153ReferenceQualifier fromValue(String v) {
        return valueOf(v);
    }

}
```

**Comment by claudemamo** [comment]
We generate the JAXB classes from DFDL schemas with _XJC_. You can customise the EDIFACT schema accordingly and then run xjc on the customised schema. I have an [example](https://github.com/claudemamo/smooks-java-to-edifact/blob/master/pom.xml#L56-L93) showing how to run XJC from a Maven project. In the project, the JAXB classes are generated from the bundled DFDL schema but you can easily change it to point to your own schema

>Also I noticed org.smooks.edifact.binding.d95b.E1153ReferenceQualifier from out of box doesn't have all specified qualifers in the specification.

You've uncovered a bug in the schema generation code. We'll address it in RC1.

**Comment by sungung** [comment]
i will try to run XJC, it will solve my problem. But as you understand, all enum type of data (all qualifiers) has only first member.

**Comment by sungung** [comment]
Hi Claude,  I can manage to populate all codes by changing the [code](https://github.com/smooks/smooks-edi-cartridge/blob/dc705c28e01cd36affebcfbb77d9de7bd387ba82/ect/src/main/java/org/smooks/edi/ect/formats/unedifact/parser/UnEdifactDirectoryParser.java#L611) as below

```
        while (line != null && !line.matches(ELEMENT_SEPARATOR)) {
            Matcher codeMatcher = CODE.matcher(line);
            if (codeMatcher.matches()) {
                codeList.getCodes().add(codeMatcher.group(2));
                while (line != null && line.length() > 0) {
                    line = reader.readLine();
                    // D9X version EDIFACT source document does't use code separator, 
                    // but fixed length indentation of each code can be used as the code separator.
                	if (line != null && line.length() > 8 && line.substring(0, 8).trim().length() > 0) {
                		break;
                	}                    
                }
            }
            line = reader.readLine();
        }
```

**Comment by sankethpb** [comment]
@sungung Kindly clarify when this could be included in the build? Thank you in advance.

**Comment by claudemamo** [comment]
This was fixed in #214.
